StyleCop.Analyzers/StyleCop.Analyzers.Test/Verifiers/DiagnosticVerifier.cs

```diff
@@ -135,71 +135,142 @@ private static void VerifyDiagnosticResults(IEnumerable<Diagnostic> actualResult
         {
             int expectedCount = expectedResults.Count();
             int actualCount = actualResults.Count();
+                StringBuilder errorStringBuilder = new StringBuilder();
 
             if (expectedCount != actualCount)
             {
-                string diagnosticsOutput = actualResults.Any() ? FormatDiagnostics(analyzer, actualResults.ToArray()) : "    NONE.";
+                errorStringBuilder.AppendLine($"Mismatch between number of diagnostics returned, expected \"{expectedCount}\" actual \"{actualCount}\"");
+            }
 
-                Assert.True(false,
-                    string.Format("Mismatch between number of diagnostics returned, expected \"{0}\" actual \"{1}\"\r\n\r\nDiagnostics:\r\n{2}\r\n", expectedCount, actualCount, diagnosticsOutput));
+            AppendMissmatchErrorText(actualResults, analyzer, expectedResults, errorStringBuilder);
+
+            string errorString = errorStringBuilder.ToString();
+            if (!string.IsNullOrEmpty(errorString))
+            {
+                Assert.True(false, errorString);
             }
```

Comment:
147-151 can be replaced by a single statement:
`Assert.True(errorStringBuilder.Length == 0, errorStringBuilder.ToString());`
or alternatively replace 148-151 with
`Assert.True(string.IsNullOrEmpty(errorString), errorString);`